### PR TITLE
Add QueueItem story and mock service worker

### LINKED ISSUE #1345: Add test queue to Storybook
Add the card and timer

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- QueueItem story + mock service work/API request mocking paradigm for storybook
- Closes #1345 

## Changes Proposed

- Adds QueueItem.stories.tsx
- More importantly, adds the mock service worker `msw` library and associated storybook addon
- Adds the ability to mock any graphql call, many of which have been mocked for the QueueItem. MSW should be used for regular fetch mocks as well but I just didn't need any yet.
- The `storyMocks.ts` file contains a list of existing mocks and a `getMocks` function to grab whatever mocks you might need for a story. It would be nice to keep the `mocks` object in this file up-to-date so we don't have to continually recreate the same mocks.

## Additional Information

- I decided to create a separate `storybook_public` directory for our storybook builds as adding the `mockServiceWorker.js` file to our `public` folder made me uneasy as it would have been included in our prod builds.

## Screenshots / Demos

Run storybook locally using `yarn storybook` and navigate to http://localhost:6006.

### UI
- [N/A] Any changes to the UI/UX are approved by design 
- [N/A] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [N/A] Database changes are submitted as a separate PR
  - [N/A] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [N/A] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
- [N/A] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [N/A] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [N/A] DevOps team has been notified if PR requires ops support
- [N/A] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: frontend/.storybook/main.js
Comment (nickscialli-usds) [inline frontend/.storybook/main.js:12]: This is how we can mock out any pesky imports for storybook

File: frontend/mockServiceWorker.js
Comment (nickscialli-usds) [inline frontend/mockServiceWorker.js:1]: this file just gets copied into the `/storybook_public` directory when starting or building storybook

File: frontend/src/stories/__mocks__/appInsights.js
Comment (nickscialli-usds) [inline frontend/src/stories/__mocks__/appInsights.js:2]: just so it doesn't explode when clicking buttons in storybook.

File: frontend/src/stories/storyMocks.tsx
Comment (nickscialli-usds) [inline frontend/src/stories/storyMocks.tsx:9]: This is the list of all mocks that I'd like to aggregate here for reuse

File: frontend/src/stories/testQueue/QueueItem.stories.tsx
Comment (nickscialli-usds) [inline frontend/src/stories/testQueue/QueueItem.stories.tsx:18]: the `getMocks` function helps us pluck out the individual mocks we want from the `storyMocks` file

File: frontend/src/stories/testQueue/QueueItem.stories.tsx
Comment (nickscialli-usds) [inline frontend/src/stories/testQueue/QueueItem.stories.tsx:20]: If your component needs graphql, make sure to add a decorator to the suite of stories or otherwise wrap the individual stories in the `StoryGraphQLProvider`

File: frontend/src/stories/testQueue/QueueItem.stories.tsx
Comment (maya) [inline frontend/src/stories/testQueue/QueueItem.stories.tsx:7]: Update to sentence case:
```suggestion
  title: "App/Test queue/Queue item",
```

**Comment by maya** [comment]
Looks awesome 🤘 I'll let the others chime in with a technical review.

**Comment by maya** [comment]
Are there any controls for changing the `COMPLETE` label to the other label options?

**Comment by maya** [comment]
@nickscialli-usds Can we add something to see the full flow from start to finish of starting the test and seeing it complete? For example, see states of: Prior to starting a test, in-progress, and complete? (or any other states we're missing?)

Note from Tim:
>Yes it would be nice to be able to see the state when a test has been completed.
It would be ideal if we could have an interactive test queue card where the timer is only a few seconds long

File: frontend/src/stories/storyMocks.tsx
Comment (nickclyde) [reply frontend/src/stories/storyMocks.tsx:9]: Will we add our REST mocks here as well?

**Review by nickclyde — APPROVED** [review]
Awesome work, this will be super useful!

File: frontend/src/stories/storyMocks.tsx
Comment (nickscialli-usds) [reply frontend/src/stories/storyMocks.tsx:9]: I believe adding the REST mocks here would work

**Comment by nickscialli-usds** [comment]
I think this is ready to go. Note that the completed queue timer state is now captured in storybook and, predictably, looks bad because it won't get fixed until #1740 lands. The good news is, if we get this current PR merged, we should be able to use it to confirm #1740 fixes the issue.

**Review by maya — APPROVED** [review]
Great work!

**Pull request merged**
